scaffolding.md

```diff
@@ -1,217 +1,280 @@
+```
+With `name:string`, `price:float`, `subject_id:integer`, and `description:text` we are giving Rails information about the fields we would like in our database table and the type of data they should accept.
+
 When you type this command, you will again see a long list of output that explains everything Rails is generating for you. The output below highlights some of the more significant things for our setup:
-   rails generate scaffold Shark name:string facts:text
-    
-   Output
- invoke active_record
-create db/migrate/20190804181822_create_sharks.rb create app/models/shark.rb
-.. .
-invoke resource_route
- route    resources :sharks
+
+```
+# Output
+invoke  active_record
+create    db/migrate/20210318152156_create_books.rb
+create    app/models/book.rb
+invoke    test_unit
+create      test/models/book_test.rb
+create      test/fixtures/books.yml
+invoke  resource_route
+route    resources :books
 invoke  scaffold_controller
-create    app/controllers/sharks_controller.rb
+create    app/controllers/books_controller.rb
 invoke    erb
-create create create create create create .. .
-app/views/sharks
-app/views/sharks/index.html.erb
-app/views/sharks/edit.html.erb
-app/views/sharks/show.html.erb
-app/views/sharks/new.html.erb
-app/views/sharks/_form.html.erb
-Rails has created the model at app/models/shark.rb and a database migration to go with it: db/migrate/20190804181822_create_sharks.rb. The timestamp on your migration file will differ from what you see here.
-It has also created a controller, app/controllers/sharks_controller.rb , as well as the views associated with our application's CRUD operations, collected under app/views/sharks . Among these views is a partial,
-, that contains code used across views.
-       _form.h
-    tml.erb
+create      app/views/books
+create      app/views/books/index.html.erb
+create      app/views/books/edit.html.erb
+create      app/views/books/show.html.erb
+create      app/views/books/new.html.erb
+create      app/views/books/_form.html.erb
+invoke    resource_route
+invoke    test_unit
+create      test/controllers/books_controller_test.rb
+create      test/system/books_test.rb
+invoke    helper
+create      app/helpers/books_helper.rb
+invoke      test_unit
+invoke    jbuilder
+create      app/views/books/index.json.jbuilder
+create      app/views/books/show.json.jbuilder
+create      app/views/books/_book.json.jbuilder
+invoke  assets
+invoke    scss
+create      app/assets/stylesheets/books.scss
+invoke  scss
+create    app/assets/stylesheets/scaffolds.scss
+```
+Rails has created the model at `app/models/book.rb` and a database migration to go with it: `db/migrate/20210318152156_create_books.rb`. The timestamp on your migration file will differ from what you see here. You can see the fields we included in our initial call of `rails generate scaffold`. Let's update the title column with a few constraints.
+```
+create_table :books do |t|
+  t.string :title, :string, limit: 32, null: false
```

Comment:
When I use `dip` to generate files in the application, it always sets the permissions wrong and I have to run the following command before I can modify them:

```
sudo chown -R nathan .
```

Did you figure out a way around that?

Reply:
@nathancolgate I saw your alias comment yesterday, do you think this is something we could declare in an introduction of the book... eg. 

_When generating files within a container you are going to need to update the permissions in order to save or edit the file. We do this with the following command_
```
sudo chown -R nathan .
```

Reply:
👍

Reply:
If everything looks good here lets merge this, and I will add it in separately. @nathancolgate